### PR TITLE
set up docker file for clion debugging, added ssh and docker compose

### PR DESCRIPTION
**Pull request by tarquasso**
added gdbserver and ssh to docker file. added docker compose file. 
Clion debugger works in this image now.

build image and run in container by:
cd drake-torch
docker-compose up -d --build

(leave out --build tag if image was already build).

Please check if the image still works for you in normal compile modes.
Clion setup I will document next.

### CODE REVIEW

File: drake-torch.dockerfile [hunk 1]
```diff
@@ -115,6 +154,10 @@ RUN apt-get update && apt-get install -y \
     iputils-ping \
     && rm -rf /var/lib/apt/lists/*
 
+# plase setup.bash script into bashrc file
+RUN touch $HOME/.bashrc \
+ && echo '/opt/ros/${ROS_DISTRO}/setup.bash' > $HOME/.bashrc
+
```
Comment (duncancmt-dexai) [inline drake-torch.dockerfile]: Shouldn't this go in `scripts/fix_bashrc.sh`?

File: drake-torch.dockerfile [hunk 2]
```diff
@@ -187,8 +230,13 @@ RUN cd $HOME && git clone https://github.com/lcm-proj/lcm.git \
...
+# setup entrypoint (not needed anymore)
+# ADD scripts/docker_entrypoint.sh $HOME/docker_entrypoint.sh
+# RUN chmod +x $HOME/docker_entrypoint.sh
+# CMD ["$HOME/docker_entrypoint.sh"]
 
-ENTRYPOINT ["docker_entrypoint.sh"]
-CMD ["bash"]
+# start ssh daemon
+CMD ["/usr/sbin/sshd", "-D"]
```
Comment (duncancmt-dexai) [inline drake-torch.dockerfile]: Why are we running `sshd` inside the docker? Can't we just use `docker exec` to get inside the docker?

File: scripts/docker_entrypoint.sh [hunk 3]
```diff
@@ -1,6 +1,4 @@
 #!/bin/bash
 set -e
-
-# setup ros environment
-source "/opt/ros/$ROS_DISTRO/setup.bash"
+# emtpy for now
```
Comment (duncancmt-dexai) [inline scripts/docker_entrypoint.sh]: Typo

File: drake-torch.dockerfile (hunk 1 above)
Comment (tarquasso) [reply drake-torch.dockerfile]: fixed

File: drake-torch.dockerfile (hunk 2 above)
Comment (tarquasso) [reply drake-torch.dockerfile]: we like to start it in detached mode. you can just use docker exec to get inside the docker.

**Pull request merged**